python-sdk/src/astro/files/base.py

```diff
@@ -124,6 +128,27 @@ def __eq__(self, other) -> bool:
     def __hash__(self) -> int:
         return hash((self.path, self.conn_id, self.filetype))
 
+    @uri.default
+    def _path_to_dataset_uri(self) -> str:
+        """Build a URI to be passed to Dataset obj introduced in Airflow 2.4"""
+        from urllib.parse import urlencode, urlparse
+
+        parsed_url = urlparse(url=self.path)
+        netloc = parsed_url.netloc
+        parsed_scheme = parsed_url.scheme
+        # Local filepaths do not have scheme
+        scheme = f"astro+{parsed_scheme}" if parsed_scheme else "astro+local"
```

Comment:
```suggestion
        scheme = f"astro+{parsed_scheme}" if parsed_scheme else "astro+file"
```
Shouldn't we use `file` for local files, following the URI convention?
https://www.iana.org/assignments/uri-schemes/uri-schemes.xhtml

Reply:
Also, what are we gaining by not representing files using their original URIs, since they are already unique?

Reply:
Done, changed in https://github.com/astronomer/astro-sdk/pull/786/commits/a783557f7c3bb532a1852d3a713955a5981dd6d3